### PR TITLE
gitserver: change RequestRepoMigrate signature

### PR DESCRIPTION
**Pull request by asdine**
This changes the gitserver client `RequestRepoMigrate` signature to take more explicit arguments.
Previously, this method was designed for the migration only and thus was determining the source and destination
based on hashing algorithms.
This change allows us to explicitly choose the source and destination.

## Test plan

Updated the unit test

Related to #32827

### CODE REVIEW

**Review by sashaostrikov — APPROVED** [review]
:shipit:

File: internal/gitserver/client.go [hunk 1]
```diff
@@ -204,9 +204,9 @@ type Client interface {
 	// UpdatedAt field will be zero. This can happen for new gitservers.
 	ReposStats(context.Context) (map[string]*protocol.ReposStats, error)
 
-	// RequestRepoMigrate is effectively RequestRepoUpdate but with some additional metadata to aid our
-	// migration of gitserver repos to the rendezvous hashing scheme.
-	RequestRepoMigrate(context.Context, api.RepoName) (*protocol.RepoUpdateResponse, error)
+	// RequestRepoMigrate is effectively RequestRepoUpdate but with some additional metadata to make
+	// gitserver instances clone a repo from one instance to another
+	RequestRepoMigrate(context.Context, api.RepoName, string, string) (*protocol.RepoUpdateResponse, error)
```
Comment (vrto) [inline internal/gitserver/client.go:209]: Should we give these names and the interface level as well? Seeing the `string, string` doesn't tell me what these are unless I look at the actual implementation.

**Review by vrto — COMMENTED** [review]
Just one small question to 🤔 about :)

File: internal/gitserver/client.go (hunk 1 above)
Comment (asdine) [reply internal/gitserver/client.go:209]: Good point, it's much clearer this way

**Review by vrto — APPROVED** [review]

**Pull request merged**